docs/development/coding-style-guide.md

```diff
@@ -0,0 +1,262 @@
+- `fmt.Errorf("space probe is offline")`
+
+Bad:
+
+- `fmt.Errorf("parsing config failed: %w", err)`
+- `fmt.Errorf("error encountered while deploying cluster: %w", err)`
+- `fmt.Errorf("could not reverse time: %w", err)`
+
+#### Capitalizing error strings
+
+Don't capitalize error strings when *returning* an error (remember - errors are chained).
+Capitalize the **beginning** of an error string when **logging** an error.
+
+Good:
+
+- `fmt.Errorf("parsing config: %w", err)`
+- `log.Errorf("Deploying cluster: %v", err)`
+
+Bad:
+
+- `fmt.Errorf("Parsing config: %w", err)`
+- `log.Errorf("deploying cluster: %v", err)`
+
+### Comments
+
+Follow the standard Go [guidelines](https://blog.golang.org/godoc) for comments in Go code. *In
+addition*, follow the guidelines below.
+
+#### Whitespace
+
+Include a space between `//` and the beginning of the comment.
+
+Good:
+
+```go
+// This is a comment.
+```
+
+Bad:
+
+```go
+//This is a comment.
+```
+
+#### Capitalization
+
+Capitalize the first letter of a Go comment unless the official Go guidelines require you to do
+otherwise.
+
+Good:
+
+```go
+// A very important constant.
+const this = "that"
+```
+
+Bad:
+
+```go
+// a very important constant.
+const this = "that"
+```
+
+#### Periods
+
+As a general rule, end all comments with a period. Possible exceptions are *very* short comments
+(up to 2-3 words which don't form a self-contained sentence) which are typically **inline**,
+however generally you should format comments as full sentences ending with a period.
+
+Good:
+
+```go
+// Foo does this and also that.
+func Foo() {
+    ...
+    n := 3 // Default
+}
+```
+
+Bad:
+
+```go
+// Foo does this and also that
+func Foo() {
+    ...
+}
+```
+
+#### "nolint" comments
+
+"nolint" comments are a special type of comment used to tell the linter to ignore a specific
+linting rule. They are formatted differently from "regular" Go comments.
+
+Format "nolint" comments according to the following example:
+
+```go
+//nolint:funlen,gosec
+func foo() {
+    ...
+}
+```
+
+- **Don't** put a space between `//` and `nolint`.
+- Don't capitalize "nolint" comments.
+- Don't end "nolint" comments with a period.
+- Don't put spaces after commas when multiple linting rules are ignored.
+
+When adding a "nolint" comment for a function which has a doc comment, put the "nolint" line below
+the doc comment after a "separator" comment line:
+
+```go
+// Foo does this and that.
+//
+//nolint:funlen,gosec
+func Foo() {
+    ...
+}
+```
+
+### Table-driven tests
+
+Whenever possible, use [table-driven tests](https://github.com/golang/go/wiki/TableDrivenTests).
+Table-driven tests are compact, readable and easy to maintain compared to test cases that are
+described imperatively.
+
+It is usually possible to construct most tests in a way which allows the various test cases to be
+described *declaratively* in a Go struct. Most tests can be broken down to "steps" similar to the
+following:
```

Comment:
Is this paragraph and all that comes after really needed?

Reply:
My intention here was to emphasize that we should try to describe tests declaratively using the table-driven tests approach. Which part do you think could be unnecessary?